### PR TITLE
Added logging

### PR DESCRIPTION
**Pull request by joandrsn**
Replaces `print` with `logging`-module.
https://docs.python.org/3/howto/logging.html#when-to-use-logging

### CODE REVIEW

File: config.ini [hunk 1]
```diff
@@ -9,8 +9,12 @@ capture = UNPARSED
...
 
 [slack]
-debug = false
+debug = true
 token = xoxb-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 channel_id = C037DRRH9TP
 threshold_percentage = 0.25
-notif_data_path = notif_data.pickle
\ No newline at end of file
+notif_data_path = notif_data.pickle
```
Comment (falkecarlsen) [inline config.ini]:
```suggestion
```

Legacy config

File: config.ini [hunk 2]
```diff
@@ -9,8 +9,12 @@ capture = UNPARSED
 location = db.sqlite3
 
 [slack]
-debug = false
+debug = true
```
Comment (falkecarlsen) [inline config.ini:12]: Sure you want to default to debug?

File: config.py [hunk 3]
```diff
@@ -31,11 +32,28 @@ def get_notif_settings():
...
+def setup_logging():
+    read_config()
+    loglevel = CONFIG.get("log", "level", fallback="DEBUG")
+    logfile = CONFIG.get("log", "filename", fallback=None)
+
+    numeric_level = getattr(logging, loglevel.upper(), None)
+    if not isinstance(numeric_level, int):
+        raise ValueError(f"Invalid log level: {loglevel}")
+    logging.basicConfig(
+        filename=logfile,
```
Comment (falkecarlsen) [inline config.py:43]: Possible `None` here, does that make logging go to stdout/err or crash?

**Review by falkecarlsen — CHANGES_REQUESTED** [review]
Small questons

File: config.ini (hunk 2 above)
Comment (joandrsn) [reply config.ini:12]: Yes to not attempt calling Slacks API with bad credentials

File: config.py (hunk 3 above)
Comment (joandrsn) [reply config.py:43]: Are you talking about `numeric_level` or `filename`?
`numeric_level` should be handled:
```python
>>> isinstance(None, int)
False
>>> isinstance(1, int)
True
```
`filename` being `None` should also be handled by the `logging`-library

**Review by falkecarlsen — APPROVED** [review]

**Pull request merged**